### DISCUSSION TITLE
How to run Lychee with the docs/api pages available [Renamed for clarity]

### DISCUSSION THREAD

**Discussion by agademer** [LycheeOrg/Lychee#2278] [Q&A] [answered]
Hi, 

first I would like to thank  you a lot for making this great tool.

I would like to access to the API documentation but it's not anymore on lycheeOrg documentation page (https://lycheeorg.github.io/docs/)

It says that since 4.8.1 it should be found as a Route /docs/api but I cannot find any route that correspond to it. :( (And I have search deeply in github)

I have found a command in CLI : php artisan route::list that list the possible routes (including the api ones)
I'm also aware of the routes/api.php file in the github.

But neither will give me an easy access to which parameter I should pass to each api route... (like I had last year with the 4.7.1 Lychee version).

Is this auto-generated doc really available ? If yes, how ?
If not, where can I find the information ?

Thanks in advance !

**Comment by ildyria** [comment]
You need to _activate_ the dev version. 
Make sure you did `composer install` instead of `composer install --no-dev` then the links will pop in the left menu.

**Comment by agademer** [comment]
Hi Ildryria,

I tried hard but I had no luck with getting a proper lychee (in dev mode) from source.

What I did :

```
sudo apt install composer
curl -sL https://deb.nodesource.com/setup_21.x | sudo bash -
sudo apt install nodejs 
sudo chmod o+w /var/www/html
sudo apt install php8.3-imagick php8.3-zip php8.3-sqlite3


git clone https://www.github.com/LycheeOrg/Lychee /var/www/html/Lychee 
composer install

sudo chown -R www-data storage
sudo chown www-data .
sudo chown www-data database
sudo chown www-data database/database.sqlite
sudo chown -R www-data public
sudo chown www-data Lychee/.env

--> Configure Lychee through install process

ex : 
APP_NAME=Lychee
APP_ENV=dev
APP_KEY=base64:MYKEY
APP_DEBUG=false
# This MUST contain the host name up to the Top Level Domain (tld) e.g. .com, .org etc.
APP_URL=https://monurl.com
APP_FORCE_HTTPS=true

DB_CONNECTION=mysql
DB_HOST=localhost
DB_PORT=3306
DB_DATABASE=lychee_db
DB_USERNAME=lychee
DB_PASSWORD=XXXXXXXXX
TIMEZONE=Europe/Paris
TRUSTED_PROXIES=*

npm install
npm run dev &
```

Accessing monurl.com/docs/api do work !!!
But monurl.com gallery is not properly showing anymore... :(

I suspect invalid connection to the npm
![image](https://github.com/LycheeOrg/Lychee/assets/7581611/4cbd63b0-85ac-4c9e-a1a7-a3c25255ef6b)
but I don't know how to fix it.
(I'm beyond a reverse proxy that only let pass 80/443, but I don't understand why those request are needed between my browser and the site)

----

I also tried to install from docker (I have seen that there is a lychee:dev image)

I deployed with :
```
docker network create --driver bridge lychee_bridge

docker run -d --name=mariadb -e MYSQL_ROOT_PASSWORD=XXX-e MYSQL_DATABASE=lychee_db -e MYSQL_USER=lychee -e MYSQL_PASSWORD=XXX--net lychee_bridge --link lychee mariadb

docker run -d --name=lychee -v /home/mde-admin/lychee_docker/conf:/conf -v /home/mde-admin/lychee_docker/uploads:/uploads -v /home/mde-admin/lychee_docker/sym:/sym -e PUID=1000 -e PGID=1000 -e PHP_TZ=Europe/Paris -e DB_CONNECTION=mysql -e DB_HOST=mariadb -e DB_PORT=3306 -e DB_DATABASE=lychee_db -e DB_USERNAME=lychee -e DB_PASSWORD=XXX --net lychee_bridge --link mariadb -p 80:80  lycheeorg/lychee:dev
```
The main website works, but no luck with docs/api pages. :( :(

Can you help me either : 
 - to have my main site work with npm run dev ?
 - to activate docs/api pages in docker.
 
 I don't understand why it is so hard to get thoses pages (a simple SHOW_API_PAGES=true in .env would so simple) and why the two mode operate so differently (from release/docker, I don't have to run any npm command... why do I have to do that in dev mode ?)
 
 Thanks in advance,
 
 Antoine

**Comment by agademer** [comment]
Hi everyone,

with a bit of perseverance.

On a fresh docker lychee:dev install (see previous message)

```
docker exec -it lychee bash
#apt install composer
#su www-data -s /bin/bash -c "composer install"
```
(The process finish with a write file error, maybe because there was a previous composer installation during the image design)
But after that, it seem that I have a functionning main site and the docs/api pages !

On a "Release" install
```
sudo apt install composer
sudo apt install php8.3-imagick php8.3-zip php8.3-sqlite3
sudo chown www-data:www-data -R /var/www/html/Lychee
cd /var/www/html/Lychee
sudo -u www-data composer install
```
(No error)
After that, it seem that I have a functionning main site and the docs/api pages !


I need to continue the tests to see if everything works.

**Reply by ildyria** [reply, marked as answer]
Hi sorry for being a bit slow.
If you are using the Lychee docker image you just picked the wrong one:

The following tags are available :

- `latest`: Latest Lychee release
- `v[NUMBER]`: Stable version tag for a Lychee release
- `nightly` (also `dev`): Current master branch tag (Lychee operates on a stable master, so this should usually be safe)
- `devtools`: As above, but includes development dependencies

### `devtools` is the one you needed.

In our case, `dev` does not means development build but more head on master.

**Reply by agademer** [reply]
So simple when you have the info X_x

It works like a charm.

If someone pass here with the same question

mkdir -p /home/$USER/lychee_docker/mysql
mkdir -p /home/$USER/lychee_docker/conf
mkdir -p /home/$USER/lychee_docker/sym
mkdir -p /home/$USER/lychee_docker/uploads

docker network create --driver bridge lychee_bridge

docker run -d --name=mariadb -v /home/$USER/lychee_docker/mysql:/var/lib/mysql -e MYSQL_ROOT_PASSWORD=xxx -e MYSQL_DATABASE=lychee_db -e MYSQL_USER=lychee -e MYSQL_PASSWORD=xxx --net lychee_bridge --link lychee mariadb

docker run -d --name=lychee -v /home/$USER/lychee_docker/conf:/conf -v /home/$USER/lychee_docker/uploads:/uploads -v /home/$USER/lychee_docker/sym:/sym -e PUID=1000 -e PGID=1000 -e PHP_TZ=Europe/Paris -e DB_CONNECTION=mysql -e DB_HOST=mariadb -e DB_PORT=3306 -e DB_DATABASE=lychee_db -e DB_USERNAME=lychee -e DB_PASSWORD=xxx --net lychee_bridge --link mariadb -p 80:80  lycheeorg/lychee:devtools

A big thanks to you ildyria !

(Maybe add this type of explaination in your README or in the api section of the doc).

**Comment by d7415** [comment]
FWIW if you just want the docs, something like `docker run -p 8080:80 lycheeorg/lychee:devtools` should work on its own (can't test right now)
